deltacat/catalog/model/catalog.py

```diff
@@ -136,46 +212,107 @@ def init(
         Catalog, serializer=Catalog.__reduce__, deserializer=Catalog.__init__
     )
 
-    global all_catalogs
+    global catalog_registry
 
-    all_catalogs = Catalogs.remote(
+    catalog_registry[namespace] = Catalogs.remote(
         catalogs=catalogs, default_catalog_name=default_catalog_name
     )
 
 
-def get_catalog(name: Optional[str] = None) -> Catalog:
-    from deltacat.catalog.model.catalog import all_catalogs
+def get_catalog(name: Optional[str] = None, **kwargs) -> Catalog:
+    """
+    Get a catalog by name or the default catalog if no name is provided.
 
-    if not all_catalogs:
-        raise ValueError(
-            "No catalogs available! Call "
+    Args:
+        name: Name of catalog to retrieve (optional, uses default if not provided)
+
+    Returns:
+        The requested Catalog, or KeyError if it does not exist
+    """
+    namespace = kwargs.get("namespace", _default_namespace)
+
+    global catalog_registry
+    if namespace not in catalog_registry:
+        raise KeyError(
+            f"No catalogs available! Call "
             "`deltacat.init(catalogs={...})` to register one or more "
             "catalogs then retry."
         )
-    catalog = (
-        ray.get(all_catalogs.get.remote(name))
-        if name
-        else ray.get(all_catalogs.default.remote())
-    )
+
+    catalog_actor = catalog_registry[namespace]
+
+    if name is not None:
+        catalog = ray.get(catalog_actor.get.remote(name))
+    else:
+        default = ray.get(catalog_actor.default.remote())
+        if not default:
+            available_catalogs = ray.get(catalog_actor.all.remote()).values()
+            raise KeyError(
+                f"Call to get_catalog without name failed because no default catalog is configured."
+                f"Available catalogs are: {available_catalogs}"
+            )
+        catalog = default
+
     if not catalog:
-        available_catalogs = ray.get(all_catalogs.all.remote()).values()
-        raise ValueError(
-            f"Catalog '{name}' not found. Available catalogs: " f"{available_catalogs}."
+        available_catalogs = ray.get(catalog_actor.all.remote()).values()
+        raise KeyError(
+            f"Catalog '{name}' not found in namespace '{namespace}'. "
+            f"Available catalogs: {available_catalogs}."
         )
     return catalog
 
 
-def put_catalog(name: str, *args, **kwargs) -> Catalog:
-    from deltacat.catalog.model.catalog import all_catalogs
+def put_catalog(
+    name: str,
+    *,
```

Comment:
What's this wildcard? is this intended?

Reply:
This marks the end of positional arguments, and looks right to me in this case (i.e., the minimal signature for invoking this method would look like `put_catalog("my_catalog")`.